xml/security_ldap_migrate_test.xml

```diff
@@ -179,4 +179,150 @@ You should now review your instance configuration and data:
+     Replication topology.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     High availability and load balancer configurations.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     External data flows (IGA, HR, AD, etc.).
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Configured overlays (plugins in &ds389;).
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Client configuration and expected server features.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Customized schema.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     TLS configuration.
+    </para>
+   </listitem>
+  </itemizedlist>
+  <para>
+   Plan what your &ds389; deployment will look like in the end. This includes
+   the same list, except replace overlays with plugins. Once you have assessed
+   your current environment, and planned what your &ds389; environment will
+   look like, you can then form a migration plan. It is recommended you build
+   the &ds389; environment in parallel to your OpenLDAP environment to allow
+   you to switch between them.
+  </para>
+  <para>
+   Migrating from OpenLDAP to &ds389; is a one-way migration. There are enough
+   differences between the two that they cannot interoperate, and there is not
+   a migration path from &ds389; to OpenLDAP. The following table highlights
+   the major similarities and differences.
+  </para>
+  <informaltable>
+   <tgroup cols="4">
+    <thead>
+     <row>
+      <entry>Feature</entry>
+      <entry>OpenLDAP</entry>
+      <entry>&ds389;</entry>
+      <entry>Compatible</entry>
+     </row>
+    </thead>
+    <tbody>
+     <row>
+      <entry>Two-way replication</entry>
+      <entry>SyncREPL</entry>
+      <entry>389-specific system</entry>
+      <entry>No</entry>
+     </row>
+     <row>
+      <entry>MemberOf</entry>
+      <entry>Overlay</entry>
+      <entry>Plugin</entry>
+      <entry>Yes, simple configs only</entry>
+     </row>
+     <row>
+      <entry>External Auth</entry>
+      <entry>Proxy</entry>
+      <entry>-</entry>
+      <entry>No</entry>
+     </row>
+     <row>
+      <entry>Active Directory Synchronisation</entry>
+      <entry>-</entry>
+      <entry>Winsync Plugin</entry>
+      <entry>No</entry>
+     </row>
+     <row>
+      <entry>Inbuilt Schema</entry>
+      <entry>OLDAP Schemas</entry>
+      <entry>389 Schemas</entry>
+      <entry>Yes, supported by migration tool</entry>
+     </row>
+     <row>
+      <entry>Custom Schema</entry>
+      <entry>OLDAP Schemas</entry>
+      <entry>389 Schemas</entry>
+      <entry>Yes, supported by migration tool</entry>
+     </row>
+     <row>
+      <entry>Database Import</entry>
+      <entry>LDIF</entry>
+      <entry>LDIF</entry>
+      <entry>Yes, supported by migration tool</entry>
+     </row>
+     <row>
+      <entry>Password hashes</entry>
+      <entry>Varies</entry>
+      <entry>Varies</entry>
+      <entry>Yes, all formats supported excluding ARGON2</entry>
```

Comment:
ARGON2 (from openldap2-contrib) 

I think is the package it's in .... I'll double check to be sure.

Reply:
SLES has the libargon2-1 package. openSUSE has argon2, python2-argon2-cffi, libargon2-1, and some others.

Reply:
Yes, but 389 doesn't support importing the argon2 hashs that openldap2-contrib can create :)

Reply:
Maybe I should be clearer. Some password storage mechanisms in openldap aren't "built in" they are in an unsupported contrib module. That includes PBKDF2, SHA2, ARGON2 etc. So openldap2-contrib will likely use openssl or libargon2 or whatever, but for openldap2 to store the ARGON2 hash, it needs the openldap2contrib module.

And if someone is using this singular ARGON2 hashtype from openldap2-contrib, 389-ds can't support importing that one alone.